src/Misc/layoutbin/run-helper.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Validate not sudo
+user_id=`id -u`
+if [ $user_id -eq 0 -a -z "$RUNNER_ALLOW_RUNASROOT" ]; then
+    echo "Must not run interactively with sudo"
+    exit 1
+fi
+
+# Run
+shopt -s nocasematch
+
+safe_sleep() {
+    if [ ! -x "$(command -v sleep)" ]; then
+        if [ ! -x "$(command -v ping)" ]; then
+            COUNT="0"
+            while [[ $COUNT != 5000 ]]; do
+                echo "SLEEP" > /dev/null
+                COUNT=$[$COUNT+1]
+            done
+        else
+            ping -c 5 127.0.0.1 > /dev/null
+        fi
+    else
+        sleep 5
+    fi
+}
+
+bin/Runner.Listener run $*
+returnCode=$?
+if [[ $returnCode == 0 ]]; then
+    echo "Runner listener exit with 0 return code, stop the service, no retry needed."
```

Comment:
make sure with ting it's safe to echo here (doesnt influence daemon negatively)

Reply:
echo to stdout should be fine...